src/Servant/QuickCheck/Internal/Predicates.hs

```diff
@@ -84,15 +84,13 @@ notLongerThan maxAllowed
 -- /Since 0.0.0.0/
 onlyJsonObjects :: ResponsePredicate
 onlyJsonObjects
-  = ResponsePredicate (\resp -> case go resp of
-      Nothing -> throw $ PredicateFailure "onlyJsonObjects" Nothing resp
-      Just () -> return ())
-    where
-      go r = do
-        ctyp <- lookup "content-type" (first foldedCase <$> responseHeaders r)
-        when ("application/json" `SBS.isPrefixOf` ctyp) $ do
-          (_ :: Object) <- decode (responseBody r)
-          return ()
+  = ResponsePredicate (\resp -> do
+        case lookup "content-type" (first foldedCase <$> responseHeaders resp) of
+            Nothing    -> return ()
+            Just ctype -> when ("application/json" `SBS.isPrefixOf` ctype) $ do
+                case (decode (responseBody resp) :: Maybe Object) of
+                    Nothing -> throw $ PredicateFailure "onlyJsonObjects" Nothing resp
+                    Just _  -> return ())
```

Comment:
I'm a bit hesitant of changing indentation style here -- it's 2 space in the rest of the codebase.

Reply:
I didn't really pay attention - I assumed there would be a formatter in the CI tbh. I certainly don't mind changing that.

Reply:
I cleaned up to have 2-space indentation style, and also cleaned up the git history fuckup I'd done (now my two commits nicely sit atop the most recent master branch here).